### ISSUE TITLE
Little race in shut down code (Prometheus 2.0)

### ISSUE THREAD

**Issue by grobie** [prometheus/prometheus#3232]
**What did you do?**

Terminated Prometheus

**What did you expect to see?**

A shutdown sequence which is a bit ordered.

**What did you see instead? Under which circumstances?**

The order of the log lines was unexpected.

**Environment**

* System information:

	insert output of `uname -srm` here

* Prometheus version:

	prometheus, version 2.0.0-beta.5.sc0 (plus HTTP Accept patch)

* Logs:

```
2017-10-02_09:45:40.88289 level=info ts=2017-10-02T09:45:40.882815813Z caller=main.go:377 msg="See you next time!"
2017-10-02_09:45:43.19105 level=info ts=2017-10-02T09:45:43.190962749Z caller=targetmanager.go:82 component="target manager" msg="Stopping target manager..."
2017-10-02_09:45:43.29225 level=info ts=2017-10-02T09:45:43.292150245Z caller=targetmanager.go:94 component="target manager" msg="Target manager stopped"
2017-10-02_09:45:43.29229 level=info ts=2017-10-02T09:45:43.292217541Z caller=manager.go:455 component="rule manager" msg="Stopping rule manager..."
2017-10-02_09:45:43.30641 level=info ts=2017-10-02T09:45:43.306347592Z caller=manager.go:461 component="rule manager" msg="Rule manager stopped"
2017-10-02_09:45:43.30643 level=info ts=2017-10-02T09:45:43.306382993Z caller=notifier.go:482 component=notifier msg="Stopping notification handler..."
```

**Comment by grobie** [comment]
@fabxc @Gouthamve I think we should at least ensure that every `Stopping ...` has a according `... stopped` line. Seems to be missing for the notification handler.

**Comment by grobie** [comment]
Fixed in #3251.
